### PR TITLE
feat(aws): Add private certificate authorities

### PR DESCRIPTION
**Pull request by adam-tylr**
#### Summary

Adds support for acmpca certificate authorities.

### CODE REVIEW

File: plugins/source/aws/resources/services/acmpca/certificate_authorities.go
Comment (bbernays) [inline plugins/source/aws/resources/services/acmpca/certificate_authorities.go:19]: The name of the service in this location has to match the name that AWS [uses here](https://github.com/cloudquery/cloudquery/blob/66eefda579565b4e76647daf36ca21070d338f59/plugins/source/aws/client/data/partition_service_region.json#L79)

so in this case the service name is `acm-pca`

File: plugins/source/aws/resources/services/acmpca/certificate_authorities.go
Comment (bbernays) [inline plugins/source/aws/resources/services/acmpca/certificate_authorities.go:59]:
```suggestion
	certAuthority := resource.Item.(types.CertificateAuthority)
```

File: plugins/source/aws/resources/services/acmpca/certificate_authorities.go
Comment (bbernays) [inline plugins/source/aws/resources/services/acmpca/certificate_authorities.go:45]:
```suggestion
	paginator := acmpca.NewListCertificateAuthoritiesPaginator(svc, nil)
```

**Review by bbernays — CHANGES_REQUESTED** [review]
Overall this looks great! A few minor tweaks to get the tests passing!

**Comment by adam-tylr** [comment]
Thanks @bbernays, `make test` is passing now

**Comment by bbernays** [comment]
@adam-tylr Can you run `go mod tidy`?

**Review by bbernays — APPROVED** [review]
LGTM! Once the tests all pass this should merge

**Comment by adam-tylr** [comment]
@bbernays Any idea about the failed check?

**Comment by bbernays** [comment]
@adam-tylr - That test is now passing 😄 ... But unfortunately there are conflicts in the go mod now 😭 ...

**Review by hermanschaaf — APPROVED** [review]

**Comment by hermanschaaf** [comment]
Thanks for this contribution @adam-tylr, looks great!

**Pull request merged**
